.github/workflows/vulnerability-scanning.yml

```diff
@@ -0,0 +1,35 @@
+on: [pull_request]
+
+jobs:
+  depchecktest:
+    runs-on: ubuntu-latest
+    name: depecheck_test
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+      - name: Call setup
+        run: ./gradlew dependencies
+      - name: Depcheck
+        uses: dependency-check/Dependency-Check_Action@main
```

Comment:
I don't think it's a good idea to use a branch like that, better pin to a commit and have Renovate update it regularly.